homeassistant/components/mqtt/__init__.py

```diff
@@ -729,10 +734,18 @@ async def async_publish(
     ) -> None:
         """Publish a MQTT message."""
         async with self._paho_lock:
-            _LOGGER.debug("Transmitting message on %s: %s", topic, payload)
-            await self.hass.async_add_executor_job(
+            msg_info = await self.hass.async_add_executor_job(
                 self._mqttc.publish, topic, payload, qos, retain
             )
+            _LOGGER.debug(
```

Comment:
Shouldn't we keep debug before doing the work, in case it breaks ?

Reply:
I want to include the "mid" in the debug message.
If it breaks there would be an exception logged instead?

Reply:
Right, but then we don't know what we did that caused the exception?